### PR TITLE
A few code health improvements

### PR DESCRIPTION
**Pull request by huitseeker**
The review should be easy and non-controversial, shout if it isn't !

I do have played with the charts generated as of this PR, but please don't take my word for it !

### CODE REVIEW

File: modules/common/src/main/scala/notebook/front/widgets/package.scala [hunk 1]
```diff
@@ -414,53 +414,38 @@ package object widgets {
     ))
   }
 
-  case class ScatterChart[C:ToPoints:Sampler](originalData:C, fields:Option[(String, String)]=None, override val sizes:(Int, Int)=(600, 400), maxPoints:Int = DEFAULT_MAX_POINTS) extends Chart[C] {
-    val (f1, f2)  = fields.getOrElse((headers(0), headers(1)))
+  trait Sequencifiable[C] { self: Chart[C] =>
```
Comment (andypetrella) [inline modules/common/src/main/scala/notebook/front/widgets/package.scala]: \o/ Thanks!

**Comment by andypetrella** [comment]
Nice one.
The IntegrationSpec doesn't pass... weird exception though :-S

File: modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala [hunk 2]
```diff
@@ -36,6 +36,7 @@ class Repl(val compilerOpts: List[String], val jars:List[String]=Nil) extends Re
 
   private var _initFinished: Boolean = false
   private var _evalsUntilInitFinished: Int = 0
+  // FG: never used !
```
Comment (vidma) [inline modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala]: ?

File: modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala [hunk 3]
```diff
@@ -237,7 +238,7 @@ class Repl(val compilerOpts: List[String], val jars:List[String]=Nil) extends Re
                   line.pathTo("$rendered")+"$", true, classLoader
                 )
 
-                val o = renderedClass2.getDeclaredField(interp.global.nme.MODULE_INSTANCE_FIELD.toString).get()
+                val o = renderedClass2.getDeclaredField(interp.global.nme.MODULE_INSTANCE_FIELD.toString).get(())
```
Comment (vidma) [inline modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala]: what is this about?

**Comment by vidma** [comment]
looks :ok_hand: . see comments.

**Comment by huitseeker** [comment]
Actually, I also get the failure for 1c2549732244062a1b0fc3158af46b1d108ac94f 

Flaky test ? I tested with : 
`sbt -Dscala.version="2.11.7" -Dspark.version="1.6.0" -Dhadoop.version="2.7.1"   '++2.11.7' test`

**Comment by andypetrella** [comment]
bloody hell... need to fix this later I guess it's a flasky test introduced, or remove this test that doesn't make any sense unless that the application can start ^^

**Comment by andypetrella** [comment]
I got rid of this test @huitseeker so you can remove it too and push back. It should be okay then to merge it :-)

File: modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala (hunk 3 above)
Comment (huitseeker) [inline modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala]: Interpolated arguments.

**Comment by huitseeker** [comment]
@andypetrella Done !

**Pull request merged**